File: packages/types/src/provider-keys.ts
export type ProviderKeyName = "heygen";

export interface ProviderKeyMetadata {
  provider: ProviderKeyName;
  keyHint: string;
  updatedAt?: string | null;
}

export interface ProviderKeysResponse {
  keys: ProviderKeyMetadata[];
}

export interface ProviderKeyUpsertRequest {
  provider: ProviderKeyName;
  apiKey: string;
}

File: apps/api/app/schemas/provider_keys.py
from typing import Literal

from app.schemas.common import ApiModel

ProviderKeyName = Literal["heygen"]


class ProviderKeyMetadata(ApiModel):
    provider: ProviderKeyName
    key_hint: str
    updated_at: str | None = None


class ProviderKeysResponse(ApiModel):
    keys: list[ProviderKeyMetadata]


class ProviderKeyUpsertRequest(ApiModel):
    provider: ProviderKeyName
    api_key: str
